lib/kcerts/ssh.go

```diff
@@ -268,8 +205,101 @@ func SignPublicKey(ca ssh.Signer, certType uint32, principals []string, ttl time
 		ValidPrincipals: principals,
 		Permissions:     ssh.Permissions{},
 	}
-	if err := cert.SignCert(rand.Reader, ca); err != nil {
+	s, err := p.Signer()
+	if err != nil {
+		return nil, err
+	}
+	if err := cert.SignCert(rand.Reader, s); err != nil {
 		return nil, err
 	}
 	return cert, nil
 }
+
+func OpenSSHEncode21559PrivateKey(key ed25519.PrivateKey) ([]byte, error) {
+	b := openSSHMarshalED25519PrivateKey(key)
+	return pem.EncodeToMemory(&pem.Block{Type: "OPENSSH PRIVATE KEY", Bytes: b}), nil
+
+}
+
+// Note: copited from https://github.com/mikesmitty/edkey/blob/master/edkey.go
+// TODO(adam): support passwords
+
+//* Writes ed25519 private keys into the new OpenSSH private key format.
+//I have no idea why this isn't implemented anywhere yet, you can do seemingly
```

Comment:
nit: space between // and comment.

This sounds very suspicious, though: `ssh.MarshalAuthorizedKey` works just fine on ed25519 keys. The function [here](https://github.com/golang/crypto/blob/83a5a9bb288b/ssh/keys.go#L281) just invokes the [code here](https://github.com/golang/crypto/blob/83a5a9bb288b/ssh/keys.go#L572).

Why do we need this?

Reply:
Those are for public keys. This is for private keys.